### PR TITLE
mes-9901-chartZIndexFix

### PR DESCRIPTION
**Pull request by RLCorp**
## Description
Fixed an issue where the chart's dataLabels would be displayed under the average line
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image

### CODE REVIEW

**Review by andrewsetterfield — CHANGES_REQUESTED** [review]
rebase and remove work from other branches

File: src/components/common/chart/chart.ts [hunk 1]
```diff
@@ -403,4 +401,43 @@ export class ChartComponent implements OnInit, OnChanges {
...
+  /**
+   * Creates graph labels for the chart.
+   *
+   * This method generates an array of point annotations for the chart based on the provided data values.
+   * Each annotation point is positioned at the corresponding label
+   * on the x-axis and has a label displaying the data value.
+   *
+   * @param {number[]} dataValues - An array of data values to be used for creating the graph labels.
+   */
+  createGraphLabels(dataValues: number[]): void {
```
Comment (andrewsetterfield) [inline src/components/common/chart/chart.ts:414]: So your looping though entries to push into another array to set graph labels?  could we not just use map?

Comment (andrewsetterfield) [reply src/components/common/chart/chart.ts:414]: i asked co-pilot

createGraphLabels(dataValues: number[]): void {
  this.graphLabels = dataValues.map((value, i) => ({
    x: this.labels[i],
    y: 0,
    marker: { size: 0 },
    label: {
      offsetY: -5,
      text: value.toString(),
      borderColor: '#FFFFFF',
      style: {
        padding: { top: -1, left: 3.5, right: 3.5 },
        fontFamily: 'system-ui, -apple-system, BlinkMacSystemFont, Roboto',
        fontSize: this.getFontSize(),
        fontWeight: 'bold',
        background: '#000000',
        color: '#FFFFFF',
      },
    },
  }));
}

Comment (RLCorp) [reply src/components/common/chart/chart.ts:414]: Changed in newest commit

**Review by andrewsetterfield — APPROVED** [review]

**Review by gittins7 — APPROVED** [review]

**Pull request merged**